### DISCUSSION TITLE
Add support for indicating mentions and replies

### DISCUSSION THREAD

**Discussion by Kabouik** [d99kris/nchat#153] [Ideas]
For groups that are very active, it would be useful to visually distinguish mentions/reactions to the user's messages, ideally with `Jump to mention/reply` (and `Jump to initial message` if the mention was a reply) keybinds.

**Comment by d99kris** [comment]
Hi @Kabouik,

Thanks for suggesting a new feature. I think it's something that maybe can be considered, but in the short term, I will probably not have time to look into this.

In any case, I encourage anyone else interested in this feature to upvote it. This will help gauge the level of interest.

**Reply by Disonantemus** [reply]
Happy with just **reactions**, ideas/proposal:
- to _make a reaction_: select message > Ctrl-s > choose with same pop-up as _add emoji_ .
- to _watch a reaction_: select message > Alt-v > show reaction inside **nchat** pop-up.

[**viu**](https://github.com/atanunq/viu) can show images and animated gifs in terminal, [imagemagick's convert](https://imagemagick.org/) can be used with formats that is not supported by viu. Has native support for [Kitty](https://sw.kovidgoyal.net/kitty/graphics-protocol.html) and [iTerm](https://iterm2.com/documentation-images.html) graphics protocol for better images, but works no problem with another terminal emulator.

This also can be used (did some tests and work) with:
- emoji.
- stickers.
- animated gifs (Tenor).

![viu_st](https://github.com/d99kris/nchat/assets/63484282/de0f528b-d0a9-4f1e-ae19-db582f1059b7)

**Comment by d99kris** [comment]
Hi - FYI:
Support for showing (and adding) reactions has now been added, see #213 for details.
Support for jumping to quoted message has also been added, see #156.

**Reply by Kabouik** [reply]
Awesome! Thanks Kris.

**Comment by Kabouik** [comment]
That's great news! I'm leaving this open if you don't mind because I think showing an `nchat`-wide notification (maybe a marker or number in the chat list) when the user's messages have been reacted to or replied to, with a way to directly go to that (those) message(s), would be useful too.

**Reply by d99kris** [reply]
Yes, it makes sense to keep this item open. 👍

**Reply by d99kris** [reply]
I'm updating the title to `Add support for indicating mentions and replies` to better reflect the remaining items.

**Reply by Kabouik** [reply]
I believe it would also be useful to add a way to view the authors of reactions. I just got one, but can't see who added it, and realize it's actually an important detail to understand if my message was seen by the primary recipient, or just by someone who agrees, I suppose the same would apply to any conversation ending with reactions.

**Reply by d99kris** [reply]
Taking note on the request, will see whether it can be added.
